main/src/cgeo/geocaching/storage/DataStore.java

```diff
@@ -162,7 +165,7 @@ public Integer call(final Cursor cursor) {
                     "cg_caches.watchlistCount";          // 42
 
     /** The list of fields needed for mapping. */
-    private static final String[] WAYPOINT_COLUMNS = { "_id", "geocode", "updated", "type", "prefix", "lookup", "name", "latitude", "longitude", "note", "own", "visited", "user_note", "org_coords_empty" };
+    private static final String[] WAYPOINT_COLUMNS = { "_id", "geocode", "updated", "type", "prefix", "lookup", "name", "latitude", "longitude", "note", "own", "visited", "user_note", "org_coords_empty", "calc_state" };
```

Comment:
Okay, this one has finally been dealt with.

The current solution is to use JSON strings instead of the SQL BLOB.  The implementation is fault tolerant in that should a JSON value not be available a default value is used.  There is no versioning information implemented for the CalcState as such but should modifications need to be made in the future it shouldn't be too hard to extend the existing framework, though hopefully this won't be necessary.

One thing that does concern me though is that the same waypoint data seems to be extracted from the database surprisingly many times.  This is not a problem with my code as such but may point to a flaw elsewhere.